hy/compiler.py

```diff
@@ -712,125 +598,96 @@ def _render_quoted_form(self, form, level):
         return imports, HyExpression([HySymbol(name),
                                       form]).replace(form), False
 
-    @builds("quote", "quasiquote")
-    @checkargs(exact=1)
-    def compile_quote(self, entries):
-        if entries[0] == "quote":
+    @special(["quote", "quasiquote"], [FORM])
+    def compile_quote(self, expr, root, arg):
+        if root == "quote":
             # Never allow unquoting
             level = float("inf")
```

Comment:
Should probably just be `Inf` here to be consistent.

Reply:
Good point.

Reply:
Changed.